guava-collections/src/test/java/org/baeldung/guava/GuavaMultiSetUnitTest.java

```diff
@@ -0,0 +1,67 @@
+package org.baeldung.guava;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class GuavaMultiSetUnitTest {
+
+    @Test
+    public void givenMultiSet_whenAddAndRemoveValues_shouldReturnCorrectCount() {
+        Multiset<String> bookStore = HashMultiset.create();
+        bookStore.add("Potter");
+        bookStore.add("Potter");
+        bookStore.add("Potter");
+
+        assertThat(bookStore.contains("Potter")).isTrue();
+        assertThat(bookStore.count("Potter")).isEqualTo(3);
+
+        bookStore.remove("Potter");
+        assertThat(bookStore.contains("Potter")).isTrue();
+        assertThat(bookStore.count("Potter")).isEqualTo(2);
+    }
+
+    @Test
+    public void givenMultiSet_whenSetCount_shouldReturnCorrectCount() {
+        Multiset<String> bookStore = HashMultiset.create();
+        bookStore.setCount("Potter", 50);
+        assertThat(bookStore.count("Potter")).isEqualTo(50);
+    }
+
+    @Test
+    public void givenMultiSet_whenSettingNegativeCount_shouldThrowException() {
+        Multiset<String> bookStore = HashMultiset.create();
+        assertThatThrownBy(() -> bookStore.setCount("Potter", -1)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void givenMultiSet_whenSettingCountWithOldCount_shouldReturnCorrectValue() {
+        Multiset<String> bookStore = HashMultiset.create();
+        assertThat(bookStore.setCount("Potter", 0, 2)).isTrue();
```

Comment:
The 0 is an edge case here. I.e. when there are none, you have to provide 0. There's a mid-level case, which is that you know the count of some present items - say _there are 6_ and you set the count on the basis of that assumption.

In fact, there are three cases:

- None present - special case 0
- We know the count correctly - case n
- We do not know the count - so we get a failure

I suggest breaking each into its own test.

Reply:
Done.